chainstate/test-suite/src/tests/orders_tests.rs

```diff
@@ -0,0 +1,664 @@
+        UtxoOutPoint,
+    },
+    primitives::{Amount, Idable},
+};
+use crypto::key::{KeyKind, PrivateKey};
+use orders_accounting::OrdersAccountingDB;
+use randomness::{CryptoRng, Rng};
+use rstest::rstest;
+use test_utils::{
+    nft_utils::random_token_issuance_v1,
+    random::{make_seedable_rng, Seed},
+};
+
+use crate::tests::helpers::{issue_token_from_block, mint_tokens_in_block};
+
+fn issue_and_mint_token(
+    rng: &mut (impl Rng + CryptoRng),
+    tf: &mut TestFramework,
+) -> (TokenId, UtxoOutPoint, UtxoOutPoint) {
+    let genesis_block_id = tf.best_block_id();
+    let issuance = TokenIssuance::V1(random_token_issuance_v1(
+        tf.chain_config(),
+        Destination::AnyoneCanSpend,
+        rng,
+    ));
+    let (token_id, _, utxo_with_change) = issue_token_from_block(
+        rng,
+        tf,
+        genesis_block_id,
+        UtxoOutPoint::new(genesis_block_id.into(), 0),
+        issuance,
+    );
+
+    let best_block_id = tf.best_block_id();
+    let (_, mint_tx_id) = mint_tokens_in_block(
+        rng,
+        tf,
+        best_block_id,
+        utxo_with_change,
+        token_id,
+        Amount::from_atoms(100_000),
+        true,
+    );
+
+    (
+        token_id,
+        UtxoOutPoint::new(mint_tx_id.into(), 0),
+        UtxoOutPoint::new(mint_tx_id.into(), 1),
+    )
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn create_order_check_storage(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut rng = make_seedable_rng(seed);
+        let mut tf = TestFramework::builder(&mut rng).build();
+
+        let (token_id, tokens_outpoint, _) = issue_and_mint_token(&mut rng, &mut tf);
+
+        let ask_amount = Amount::from_atoms(rng.gen_range(1u128..1000));
+        let give_amount = Amount::from_atoms(rng.gen_range(1u128..1000));
+        let order_data = OrderData::new(
+            Destination::AnyoneCanSpend,
+            OutputValue::Coin(ask_amount),
+            OutputValue::TokenV1(token_id, give_amount),
+        );
+
+        let order_id = make_order_id(&tokens_outpoint);
+        let tx = TransactionBuilder::new()
+            .add_input(tokens_outpoint.into(), InputWitness::NoSignature(None))
+            .add_output(TxOutput::AnyoneCanTake(order_data.clone()))
+            .build();
+        tf.make_block_builder().add_transaction(tx).build_and_process(&mut rng).unwrap();
+
+        assert_eq!(
+            Some(order_data),
+            tf.chainstate.get_order_data(&order_id).unwrap()
+        );
+        assert_eq!(
+            Some(ask_amount),
+            tf.chainstate.get_order_ask_balance(&order_id).unwrap()
+        );
+        assert_eq!(
+            Some(give_amount),
+            tf.chainstate.get_order_give_balance(&order_id).unwrap()
+        );
+    });
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn create_two_orders_same_tx(#[case] seed: Seed) {
```

Comment:
I think this can be broken into two tests. One with the same order twice in two different transactions, and one with two difference orders in one transaction.

Reply:
This doesn't seem to be done

Reply:
there are two tests now: create_two_2_orders_same_tx and create_two_2_orders_same_block. Isn't it what you suggested?

Reply:
ok I see